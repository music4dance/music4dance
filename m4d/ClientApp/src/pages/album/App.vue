<script setup lang="ts">
import PageFrame from "@/components/PageFrame.vue";
import SongTable from "@/components/SongTable.vue";
import AdminFooter from "@/components/AdminFooter.vue";
import { AlbumModel } from "@/models/AlbumModel";
import { TypedJSON } from "typedjson";
import { useSongSelector } from "@/composables/useSongSelector";
import { computed } from "vue";

declare const model_: string;

const model = TypedJSON.parse(model_, AlbumModel)!;

const { songs: selected, select: selectSong } = useSongSelector();

const artistRef = computed(() => `/song/artist?name=${encodeURIComponent(model.artist || "")}`);
const hidden = computed(() => (model.artist ? ["Artist"] : []));
</script>

<template>
  <PageFrame id="app">
    <h1>
      Album: {{ model.title }}
      <span v-if="model.artist"
        >by <a :href="artistRef">{{ model.artist }}</a></span
      >
    </h1>
    <SongTable
      :histories="model.histories!"
      :filter="model.filter"
      :hide-sort="true"
      :hidden-columns="hidden"
      @song-selected="selectSong"
    ></SongTable>
    <AdminFooter :model="model" :selected="selected"></AdminFooter>
  </PageFrame>
</template>
