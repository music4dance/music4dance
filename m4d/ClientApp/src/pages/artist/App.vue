<script setup lang="ts">
import { ArtistModel } from "@/models/ArtistModel";
import { SongFilter } from "@/models/SongFilter";
import { safeDanceDatabase } from "@/helpers/DanceEnvironmentManager";
import { TypedJSON } from "typedjson";
import { Song } from "@/models/Song";
import { useSongSelector } from "@/composables/useSongSelector";

declare const model_: string;

const model = TypedJSON.parse(model_, ArtistModel)!;

const danceDB = safeDanceDatabase();
const { songs: selected, select: selectSong } = useSongSelector();

const title = `Artist: ${model.artist}`;
const filter = new SongFilter();
const histories = model.histories ?? [];
const songs = histories.map((h) => Song.fromHistory(h));
const danceIds = [
  ...songs.reduce((acc, h) => {
    const ratings = h.danceRatings;
    if (ratings) {
      ratings.forEach((r) => acc.add(r.danceId));
    }
    return acc;
  }, new Set<string>()),
];
const dances = danceIds.map((id) => danceDB.dances.find((d) => d.id === id)!);
</script>

<template>
  <PageFrame id="app" :title="title">
    <p>
      {{ histories.length }} {{ model.artist }} songs with suggestions for dances to dance to them:
      <span v-for="(dance, idx) in dances" :key="dance.id"
        ><span v-if="idx + 1 === dances.length && dances.length > 1"> and </span
        ><a :href="'/dances/' + dance.seoName">{{ dance.name }}</a
        ><span v-if="idx + 1 < dances.length && dances.length > 2">, </span></span
      >
    </p>
    <SongTable
      :histories="histories"
      :filter="filter"
      :hide-sort="true"
      :hidden-columns="['artist', 'length', 'track']"
      @song-selected="selectSong"
    ></SongTable>
    <AdminFooter :model="model" :selected="selected"></AdminFooter>
  </PageFrame>
</template>
