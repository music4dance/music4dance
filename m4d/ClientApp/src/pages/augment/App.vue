<script setup lang="ts">
import PageFrame from "@/components/PageFrame.vue";
import { AugmentModel } from "@/models/AugmentModel";
import { Song } from "@/models/Song";
import { SongDetailsModel } from "@/models/SongDetailsModel";
import { SongFilter } from "@/models/SongFilter";
import { SongHistory } from "@/models/SongHistory";
import SongCore from "@/pages/song/components/SongCore.vue";
import { TypedJSON } from "typedjson";
import AugmentInfo from "./components/AugmentInfo.vue";
import AugmentLookup from "./components/AugmentLookup.vue";
import AugmentSearch from "./components/AugmentSearch.vue";
import { getMenuContext } from "@/helpers/GetMenuContext";

import { computed, ref } from "vue";

enum AugmentPhase {
  lookup = "lookup",
  results = "results",
  edit = "edit",
}

declare const model_: AugmentModel;

const context = getMenuContext();

const model = TypedJSON.parse(model_, AugmentModel) ?? {};
const phase = ref<AugmentPhase>(AugmentPhase.lookup);
const songModel = ref<SongDetailsModel | null>(null);
const lastSong = ref<Song | null>(null);
const created = ref(false);
const propertiesString = ref("");
const tabIndex = ref(model.id ? 1 : 0);

const canAugment = computed(() => !!context.userName);

const editSong = (model: SongDetailsModel): void => {
  songModel.value = new SongDetailsModel({
    created: !!model.created,
    songHistory: model.songHistory,
    filter: new SongFilter(),
    userName: context.userName,
  });
  phase.value = AugmentPhase.edit;
};

const adminCreate = async (): Promise<void> => {
  songModel.value = new SongDetailsModel({
    created: true,
    songHistory: SongHistory.fromString(propertiesString.value),
    filter: new SongFilter(),
    userName: context.userName,
  });
  propertiesString.value = "";
  phase.value = AugmentPhase.edit;
};

const reset = (saved: boolean): void => {
  if (saved) {
    lastSong.value = Song.fromHistory((songModel.value! as SongDetailsModel).songHistory);
    created.value = !!songModel.value!.created;
  }
  phase.value = AugmentPhase.lookup;
  model.id = "";
  songModel.value = null;
};
</script>

<template>
  <PageFrame id="app">
    <BRow v-if="phase === 'lookup'">
      <BCol>
        <BAlert v-if="lastSong" dismissible show="10">
          Thank you for {{ created ? "adding" : "editing" }} <i>{{ lastSong.title }}</i> by
          {{ lastSong.artist }}
        </BAlert>
        <h1>Add Song</h1>
        <BTabs v-if="canAugment" v-model="tabIndex" card>
          <BTab title="by Title"
            ><BCardText><AugmentSearch @edit-song="editSong($event)"></AugmentSearch></BCardText
          ></BTab>
          <BTab title="by Id">
            <BCardText
              ><AugmentLookup :id="model.id!" @edit-song="editSong($event)"></AugmentLookup
            ></BCardText>
          </BTab>
          <BTab v-if="context.isAdmin" title="Admin">
            <BCardText>
              <p>Paste in text version of property list</p>
              <BInputGroup prepend="Properties">
                <BFormInput
                  id="admin-properties"
                  v-model="propertiesString"
                  palceholder="Song Properties TSV"
                  trim
                ></BFormInput>
                <BButton variant="primary" @click="adminCreate">Create</BButton>
              </BInputGroup>
            </BCardText>
          </BTab>
        </BTabs>
        <AugmentInfo v-else :id="model.id!"> </AugmentInfo>
      </BCol>
    </BRow>
    <div v-else>
      <BAlert show variant="success">
        <div v-if="songModel && songModel.created">
          <b>Create Song:</b> This song is new to music4dance, please fill in missing fields and
          click <b>Add Song</b> to add this song to the catalog. Remember, you must vote on at least
          one dance to add this song.
        </div>
        <div v-else>
          <b>Edit Song:</b> We found this song in the music4dance catalog, please vote on dance
          styles and add tags to improve the catalog.
        </div>
      </BAlert>
      <SongCore
        :model="songModel as SongDetailsModel"
        :start-editing="true"
        :creating="!!songModel && (songModel as SongDetailsModel).created"
        @song-saved="reset(true)"
        @cancel-changes="reset(false)"
      ></SongCore>
    </div>
    <BRow
      ><BCol>
        <BAlert show variant="warning">
          Adding songs is a new feature so if you run into bugs or have suggestions for improving
          this feature please don't hesitate to send email to
          <a href="mailto:info@music4dance.net">info@music4dance.net</a> or fill out our
          <a href="https://music4dance.blog/feedback/">feedback form</a> and we'll be happy to take
          your input as we contineu to improve the feature.
        </BAlert>
      </BCol>
    </BRow>
  </PageFrame>
</template>

<style lang="scss" scoped>
#service-id {
  width: 25rem;
}
</style>
