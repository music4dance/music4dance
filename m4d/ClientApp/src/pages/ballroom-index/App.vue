<script setup lang="ts">
import BallroomList from "@/components/BallroomList.vue";
import BlogTagLink from "@/components/BlogTagLink.vue";
import CompetitionCategoryTable from "@/components/CompetitionCategoryTable.vue";
import PageFrame from "@/components/PageFrame.vue";
import TempiLink from "@/components/TempiLink.vue";
import { type BreadCrumbItem, danceTrail } from "@/models/BreadCrumbItem";
import { CompetitionCategory, CompetitionGroup } from "@/models/Competition";
import { TypedJSON } from "typedjson";

declare const model_: string;
const group = TypedJSON.parse(model_, CompetitionGroup)!;
const breadcrumbs: BreadCrumbItem[] = [...danceTrail, { text: "Ballroom", active: true }];

function categoryLink(category: CompetitionCategory): string {
  return `/dances/${category.canonicalName}`;
}
</script>

<template>
  <PageFrame id="app" title="Competition Ballroom Dancing" :breadcrumbs="breadcrumbs">
    <BallroomList>
      One of the
      <a
        href="https://music4dance.blog/question-1-im-learning-to-cha-cha-where-is-some-great-music-for-practicing/"
        target="_blank"
      >
        core ideas</a
      >
      behind <a href="https://www.music4dance.net">music4dance</a> is to find interesting music to
      build playlists for competitions rounds. This page is a central location to start a search for
      songs for your own playlists based on the tempo definitions of competition ballroom dance
      established by
      <a href="http://www.worlddancesport.org/Rule/Athlete/Competition" target="_blank"
        >World Dance Council</a
      >
      and
      <a href="https://www.ndca.org/pages/ndca_rule_book/Default.asp" target="_blank"
        >National Dance Council of America</a
      >
      and our database of songs.
    </BallroomList>
    <div v-for="category in group.categories" :key="category.name">
      <h2>
        <a :href="categoryLink(category)">{{ category.name }}</a>
      </h2>
      <CompetitionCategoryTable
        :dances="category.round"
        :title="category.fullRoundTitle"
        :use-full-name="false"
      ></CompetitionCategoryTable>
    </div>
    <div>
      <TempiLink />
      <BlogTagLink title="Ballroom" tag="ballroom"></BlogTagLink>
    </div>
  </PageFrame>
</template>
