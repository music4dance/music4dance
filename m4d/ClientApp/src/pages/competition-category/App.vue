<script setup lang="ts">
import BallroomList from "@/components/BallroomList.vue";
import BlogTagLink from "@/components/BlogTagLink.vue";
import CompetitionCategoryTable from "@/components/CompetitionCategoryTable.vue";
import LinkCategory from "@/components/LinkCategory.vue";
import PageFrame from "@/components/PageFrame.vue";
import TempiLink from "@/components/TempiLink.vue";
import { ballroomTrail, type BreadCrumbItem } from "@/models/BreadCrumbItem";
import { CompetitionCategory, CompetitionGroupModel } from "@/models/Competition";
import { TypedJSON } from "typedjson";
import { computed } from "vue";

declare const model_: string;
const model = TypedJSON.parse(model_, CompetitionGroupModel)!;
const breadcrumbs: BreadCrumbItem[] = [
  ...ballroomTrail,
  { text: model?.currentCategory.name ?? "Error", active: true },
];

const category = computed(() => model?.currentCategory ?? new CompetitionCategory());

const groupTitle = computed(() => `competition ${model?.group.name.toLowerCase()} dancing`);
const groupLink = computed(
  () => `/dances/${model?.group.name.toLowerCase()}-competition-categories`,
);
</script>

<template>
  <PageFrame id="app" :title="category.name" :breadcrumbs="breadcrumbs">
    <BallroomList :name="category.name"></BallroomList>
    <CompetitionCategoryTable
      :dances="category.round"
      :title="category.fullRoundTitle"
      :use-full-name="false"
    ></CompetitionCategoryTable>
    <CompetitionCategoryTable
      v-if="category.extras && category.extras.length > 0"
      :dances="category.extras"
      :title="category.extraDancesTitle"
      :use-full-name="false"
    ></CompetitionCategoryTable>
    <div>
      Other categories of <a :href="groupLink">{{ groupTitle }}</a> are:
      <ul>
        <li v-for="cat in model.otherCategories" :key="cat.name">
          <LinkCategory :name="cat.name"></LinkCategory>
        </li>
      </ul>
    </div>
    <dl>
      <TempiLink></TempiLink>
      <BlogTagLink title="Ballroom" tag="ballroom"></BlogTagLink>
    </dl>
  </PageFrame>
</template>
