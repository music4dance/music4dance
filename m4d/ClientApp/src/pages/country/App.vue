<script setup lang="ts">
import PageFrame from "@/components/PageFrame.vue";
import { type BreadCrumbItem, danceTrail } from "@/models/BreadCrumbItem";

const breadcrumbs: BreadCrumbItem[] = [...danceTrail, { text: "Country", active: true }];
</script>

<template>
  <PageFrame id="app" title="Country Western Dance Music" :breadcrumbs="breadcrumbs">
    <p>
      Here at music4dance, we constantly seek ways to serve the broader dance community. We're
      considering adding more Country Western dances to our catalog, but we aren't experts in those
      dances. So please <a href="https://music4dance.blog/feedback/">get in touch with us</a> if
      you're a Country Western dancer and would like to see this site provide support for those
      dances in a way similar to how we support Ballroom dances, including a
      <a href="https://www.music4dance.net/dances/ballroom-competition-categories"
        >competition tempo page</a
      >, pages for each of the dances (as we currently have for
      <a href="https://www.music4dance.net/dances/country-two-step">Country Two Step</a>) and
      include those dances in the
      <a href="https://www.music4dance.net/home/counter">tempo counter</a> and
      <a href="https://www.music4dance.net/home/tempi">tempo list</a> pages.
    </p>
    <p>
      We'd especially <a href="https://music4dance.blog/feedback/">like to hear from you</a> if you
      are an expert in Country Western dances and would be willing to help us build these pages.
      We're looking to start from the tempo guidance the
      <a
        href="https://ucwdc.org/wp-content/uploads/2023/01/UCWDC-ProAm-ProPro-and-Couples-Competition-Music-BPMs-2023-2025.pdf"
        >United Country Western Dance Council</a
      >
      and the
      <a href="https://www1.worldcdf.com/pdf/RULES_BOOK.pdf">World Country Dance Federation</a>
      provide, but would be happy to have your input as well.
    </p>
  </PageFrame>
</template>
