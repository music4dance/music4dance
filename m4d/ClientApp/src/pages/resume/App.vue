<script setup lang="ts">
import { type BreadCrumbItem, infoTrail } from "@/models/BreadCrumbItem";

const breadcrumbs: BreadCrumbItem[] = [
  ...infoTrail,
  { text: "About", href: "/home/about" },
  { text: "Resume", active: true },
];
</script>

<template>
  <PageFrame id="app" title="David's Resume" :breadcrumbs="breadcrumbs">
    <p>
      <a href="/Content/DavidGray-Resume.pdf" target="_blank">Printable Version</a>
    </p>
    <div class="title">David W. Gray</div>
    <div class="section">Skills Profile</div>
    <ul>
      <li>Leadership experience using both Agile and Waterfall methodologies</li>
      <li>Mentor and coach</li>
      <li>C#/.Net, C/C++, JavaScript, TypeScript, Vue.js, MySql, T-SQL, Azure AI Search</li>
      <li>Full Stack Web, Desktop Application, Diagnostic Tools Development</li>
    </ul>
    <div class="section">Employment History</div>
    <div class="job">
      <div class="job-row">
        <div class="job-title">Founder, music4dance.net</div>
        <div class="dates">2014 — Present</div>
      </div>
      <div class="location">Seattle, WA</div>
      <div class="description">
        <ul>
          <li>
            Built a ratings and referral website that helps dancers find music that inspires them to
            dance: A Web application written in ASP.NET Core with a Vue.js front end, running in the
            Azure cloud and built around Azure AI search.
          </li>
          <li>
            This project provides me with the opportunity to work with new technologies and explore
            upgrade paths in a fully controlled sandbox. I originally developed music4dance in
            ASP.NET MVC 3 and Microsoft T-SQL. Early on, I started sprinkling knockout.js code to
            make the UI more reactive. I've kept up with the .NET upgrades, including the major .NET
            core shift, converted the core Database to Azure Cognitive Search, and all the pages
            that require a significant amount of JavaScript to Vue.js/Typescript.
          </li>
        </ul>
      </div>
    </div>
    <div class="job">
      <div class="job-row">
        <div class="job-title">
          Maintainer,
          <a href="https://github.com/bootstrap-vue-next/bootstrap-vue-next" target="_blank"
            >BootstrapVueNext</a
          >
        </div>
        <div class="dates">2025 — Present</div>
      </div>
      <div class="location">Remote</div>
      <div class="description">
        <ul>
          <li>
            When the maintainers of the front-end library that music4dance depends on (<a
              href="https://bootstrap-vue.org/"
              target="_blank"
              >bootstrap-vue</a
            >) stepped back, I joined the engineers working on a replacement. I've contributed to
            the architecture, made numerous bug fixes, added and reworked several components, built
            out the documentation, and developed a parity report to track our progress. Using AI to
            web-scrape the bootstrap-vue documentation to generate the parity report and writing a
            VitePress Plugin/Markdown Extension to generate running sample code and annotated
            samples from the same source in the documentation were just two of the more technically
            interesting challenges.
          </li>
        </ul>
      </div>
    </div>
    <div class="job">
      <div class="job-row">
        <div class="job-title">Principal Lead Software Engineer, Microsoft</div>
        <div class="dates">2003 — 2012</div>
      </div>
      <div class="location">Redmond, WA</div>
      <div class="description">
        <ul>
          <li>
            Engineering lead for the Visual Studio Debugger, Profiler, and IntelliTrace. Drove
            best-of-class diagnostics tools through multiple product cycles. Attacked many of the
            most complex problems developers face, including reducing the "not reproducible" problem
            and diagnosing issues in the cloud, desktop, and devices.
          </li>
          <li>
            Collaborated with cross-functional teams across Microsoft, including Visual Studio, C#,
            VB, C++, .NET, SharePoint, and Windows to enable features that were business critical to
            those teams and to enable the tools that my team owned to work effectively with those
            platforms and languages.
          </li>
          <li>Worked with management and the team to transition to a SCRUM process.</li>
        </ul>
      </div>
    </div>
    <div class="job">
      <div class="job-row">
        <div class="job-title">President, theGray.com</div>
        <div class="dates">1999 — 2003</div>
      </div>
      <div class="location">Seattle, WA</div>
      <div class="description">
        General software consulting, including technical feasibility analysis for venture
        capitalists, website design and deployment, custom office automation and billing software
        for a clinic, web-based survey, and data cleaning.
      </div>
    </div>
    <div class="job">
      <div class="job-row">
        <div class="job-title">Vice President of Technology, Sunhawk.com</div>
        <div class="dates">1996 — 1998</div>
      </div>
      <div class="location">Seattle, WA</div>
      <div class="description">
        Managed a cross-functional team of Software Engineers, Graphics Artists, and Music
        Librarians to build an e-commerce solution for browsing and purchasing sheet music over the
        internet. This solution included a website engine that both compiled static pages served by
        a *nix server and an IIS plugin to dynamically serve pages based on a custom music database,
        as well as a system to download encrypted sheet music and purchase the encryption key by
        processing credit card transactions.
      </div>
    </div>
    <div class="job">
      <div class="job-row">
        <div class="job-title">Lead Software Engineer, Microsoft</div>
        <div class="dates">1995 — 1996</div>
      </div>
      <div class="location">Redmond, WA</div>
      <div class="description">
        Led a team that translated some of the most compelling ideas from Visual Basic Enterprise
        Edition, including database management and component wizards, into the first Enterprise
        Edition of Visual C++.
      </div>
    </div>
    <div class="job">
      <div class="job-row">
        <div class="job-title">Senior Software Engineer, Microsoft</div>
        <div class="dates">1989 — 1995</div>
      </div>
      <div class="location">Redmond, WA</div>
      <div class="description">
        Designed and implemented the core debugger to enable it to target multiple operating systems
        running different microprocessors, locally and remotely while hosted in different
        development environments.
      </div>
    </div>
    <div class="section">Education</div>
    <ul>
      <li>B.S. Applied Mathematics and Computer Science, Carnegie Mellon University, Pittsburgh</li>
      <li>User Experience and Design Certificate, General Assembly, Seattle</li>
    </ul>
    <div class="section">Teaching/Volunteering</div>
    <ul>
      <li>
        Mentor, Computer science students in non-traditional colleges,
        <a href="https://mentorsintech.com/mentors" target="_blank">MinT</a>
      </li>
      <li>
        Team lead for student capstone project,
        <a href="https://mentorsintech.com/open-source-1" target="_blank">CodeDay.org</a>,
        <a href="https://www.codeday.org/volunteer" target="_blank">CodeDay.org</a>, Green River
        College
      </li>
      <li>
        Mentor, MicroInternship
        <a href="https://mentorsintech.com/open-source-1" target="_blank">CodeDay.org</a>
      </li>
      <li>
        General Volunteering, Website and SEO consulting,
        <a href="https://hugohouse.org/" target="_blank">Hugo House</a>
      </li>
      <li>Instructor, Discrete Math and Data Structures, Northeastern University, Seattle</li>
      <li>Volunteer Teacher, Introduction to Computer Science, Truman High School, Federal Way</li>
      <li>
        Instructor, Introduction to Programming Workshop, General Assembly, Seattle, Washington
      </li>
    </ul>
    <div class="section">Writing</div>
    <ul>
      <li>
        Technical Blog:
        <a href="https://medium.com/@dwgray/" target="_blank">https://medium.com/@dwgray/</a>
      </li>
      <li>
        Marketing Blog:
        <a href="https://music4dance.blog/" target="_blank">https://music4dance.blog/</a>
      </li>
    </ul>
  </PageFrame>
</template>

<style lang="scss" scoped>
.title {
  font-size: 3rem;
  font-weight: bold;
}
.section {
  font-size: 2rem;
  font-weight: lighter;
}
.job-title {
  font-weight: bold;
  font-size: 1.25em;
}
.job-row {
  display: flex;
  justify-content: space-between;
}
.location {
  font-weight: lighter;
}
.dates {
  font-weight: lighter;
}
.job {
  margin-bottom: 0.5em;
}
.description {
  font-size: 1.25em;
}
li {
  list-style-type: square;
  font-size: 1.25em;
}
p {
  font-size: 1.25em;
}
</style>
