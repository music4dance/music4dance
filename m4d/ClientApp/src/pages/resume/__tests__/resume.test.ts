import { describe, test } from "vitest";
import { testPageSnapshot } from "@/helpers/TestHelpers";
import App from "../App.vue";

describe("Resume", () => {
  test("Renders Resume Page", () => {
    testPageSnapshot(App);
  });
});
