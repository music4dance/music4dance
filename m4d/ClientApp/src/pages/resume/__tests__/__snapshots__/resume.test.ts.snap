// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Resume > Renders Resume Page 1`] = `
"<div data-v-0eba368d=""><span context="[object Object]">MainMenu</span>
  <nav aria-label="breadcrumb">
    <nav aria-label="breadcrumb" style="padding: 0.25rem 0.5rem;">
      <ol class="breadcrumb">
        <li class="breadcrumb-item"><a class="" href="/">music4dance</a></li>
        <li class="breadcrumb-item"><a class="" href="/home/info">Info</a></li>
        <li class="breadcrumb-item"><a class="" href="/home/about">About</a></li>
        <li class="breadcrumb-item active"><span aria-current="location">Resume</span></li>
      </ol>
    </nav>
  </nav>
  <div id="body-content" class="container-fluid body-content">
    <h1>David's Resume</h1>
    <div>
      <p data-v-0eba368d=""><a data-v-0eba368d="" href="/Content/DavidGray-Resume.pdf" target="_blank">Printable Version</a></p>
      <div data-v-0eba368d="" class="title">David W. Gray</div>
      <div data-v-0eba368d="" class="section">Skills Profile</div>
      <ul data-v-0eba368d="">
        <li data-v-0eba368d="">Leadership experience using both Agile and Waterfall methodologies</li>
        <li data-v-0eba368d="">Mentor and coach</li>
        <li data-v-0eba368d="">C#/.Net, C/C++, JavaScript, TypeScript, Vue.js, MySql, T-SQL, Azure AI Search</li>
        <li data-v-0eba368d="">Full Stack Web, Desktop Application, Diagnostic Tools Development</li>
      </ul>
      <div data-v-0eba368d="" class="section">Employment History</div>
      <div data-v-0eba368d="" class="job">
        <div data-v-0eba368d="" class="job-row">
          <div data-v-0eba368d="" class="job-title">Founder, music4dance.net</div>
          <div data-v-0eba368d="" class="dates">2014 — Present</div>
        </div>
        <div data-v-0eba368d="" class="location">Seattle, WA</div>
        <div data-v-0eba368d="" class="description">
          <ul data-v-0eba368d="">
            <li data-v-0eba368d=""> Built a ratings and referral website that helps dancers find music that inspires them to dance: A Web application written in ASP.NET Core with a Vue.js front end, running in the Azure cloud and built around Azure AI search. </li>
            <li data-v-0eba368d=""> This project provides me with the opportunity to work with new technologies and explore upgrade paths in a fully controlled sandbox. I originally developed music4dance in ASP.NET MVC 3 and Microsoft T-SQL. Early on, I started sprinkling knockout.js code to make the UI more reactive. I've kept up with the .NET upgrades, including the major .NET core shift, converted the core Database to Azure Cognitive Search, and all the pages that require a significant amount of JavaScript to Vue.js/Typescript. </li>
          </ul>
        </div>
      </div>
      <div data-v-0eba368d="" class="job">
        <div data-v-0eba368d="" class="job-row">
          <div data-v-0eba368d="" class="job-title"> Maintainer, <a data-v-0eba368d="" href="https://github.com/bootstrap-vue-next/bootstrap-vue-next" target="_blank">BootstrapVueNext</a></div>
          <div data-v-0eba368d="" class="dates">2025 — Present</div>
        </div>
        <div data-v-0eba368d="" class="location">Remote</div>
        <div data-v-0eba368d="" class="description">
          <ul data-v-0eba368d="">
            <li data-v-0eba368d=""> When the maintainers of the front-end library that music4dance depends on (<a data-v-0eba368d="" href="https://bootstrap-vue.org/" target="_blank">bootstrap-vue</a>) stepped back, I joined the engineers working on a replacement. I've contributed to the architecture, made numerous bug fixes, added and reworked several components, built out the documentation, and developed a parity report to track our progress. Using AI to web-scrape the bootstrap-vue documentation to generate the parity report and writing a VitePress Plugin/Markdown Extension to generate running sample code and annotated samples from the same source in the documentation were just two of the more technically interesting challenges. </li>
          </ul>
        </div>
      </div>
      <div data-v-0eba368d="" class="job">
        <div data-v-0eba368d="" class="job-row">
          <div data-v-0eba368d="" class="job-title">Principal Lead Software Engineer, Microsoft</div>
          <div data-v-0eba368d="" class="dates">2003 — 2012</div>
        </div>
        <div data-v-0eba368d="" class="location">Redmond, WA</div>
        <div data-v-0eba368d="" class="description">
          <ul data-v-0eba368d="">
            <li data-v-0eba368d=""> Engineering lead for the Visual Studio Debugger, Profiler, and IntelliTrace. Drove best-of-class diagnostics tools through multiple product cycles. Attacked many of the most complex problems developers face, including reducing the "not reproducible" problem and diagnosing issues in the cloud, desktop, and devices. </li>
            <li data-v-0eba368d=""> Collaborated with cross-functional teams across Microsoft, including Visual Studio, C#, VB, C++, .NET, SharePoint, and Windows to enable features that were business critical to those teams and to enable the tools that my team owned to work effectively with those platforms and languages. </li>
            <li data-v-0eba368d="">Worked with management and the team to transition to a SCRUM process.</li>
          </ul>
        </div>
      </div>
      <div data-v-0eba368d="" class="job">
        <div data-v-0eba368d="" class="job-row">
          <div data-v-0eba368d="" class="job-title">President, theGray.com</div>
          <div data-v-0eba368d="" class="dates">1999 — 2003</div>
        </div>
        <div data-v-0eba368d="" class="location">Seattle, WA</div>
        <div data-v-0eba368d="" class="description"> General software consulting, including technical feasibility analysis for venture capitalists, website design and deployment, custom office automation and billing software for a clinic, web-based survey, and data cleaning. </div>
      </div>
      <div data-v-0eba368d="" class="job">
        <div data-v-0eba368d="" class="job-row">
          <div data-v-0eba368d="" class="job-title">Vice President of Technology, Sunhawk.com</div>
          <div data-v-0eba368d="" class="dates">1996 — 1998</div>
        </div>
        <div data-v-0eba368d="" class="location">Seattle, WA</div>
        <div data-v-0eba368d="" class="description"> Managed a cross-functional team of Software Engineers, Graphics Artists, and Music Librarians to build an e-commerce solution for browsing and purchasing sheet music over the internet. This solution included a website engine that both compiled static pages served by a *nix server and an IIS plugin to dynamically serve pages based on a custom music database, as well as a system to download encrypted sheet music and purchase the encryption key by processing credit card transactions. </div>
      </div>
      <div data-v-0eba368d="" class="job">
        <div data-v-0eba368d="" class="job-row">
          <div data-v-0eba368d="" class="job-title">Lead Software Engineer, Microsoft</div>
          <div data-v-0eba368d="" class="dates">1995 — 1996</div>
        </div>
        <div data-v-0eba368d="" class="location">Redmond, WA</div>
        <div data-v-0eba368d="" class="description"> Led a team that translated some of the most compelling ideas from Visual Basic Enterprise Edition, including database management and component wizards, into the first Enterprise Edition of Visual C++. </div>
      </div>
      <div data-v-0eba368d="" class="job">
        <div data-v-0eba368d="" class="job-row">
          <div data-v-0eba368d="" class="job-title">Senior Software Engineer, Microsoft</div>
          <div data-v-0eba368d="" class="dates">1989 — 1995</div>
        </div>
        <div data-v-0eba368d="" class="location">Redmond, WA</div>
        <div data-v-0eba368d="" class="description"> Designed and implemented the core debugger to enable it to target multiple operating systems running different microprocessors, locally and remotely while hosted in different development environments. </div>
      </div>
      <div data-v-0eba368d="" class="section">Education</div>
      <ul data-v-0eba368d="">
        <li data-v-0eba368d="">B.S. Applied Mathematics and Computer Science, Carnegie Mellon University, Pittsburgh</li>
        <li data-v-0eba368d="">User Experience and Design Certificate, General Assembly, Seattle</li>
      </ul>
      <div data-v-0eba368d="" class="section">Teaching/Volunteering</div>
      <ul data-v-0eba368d="">
        <li data-v-0eba368d=""> Mentor, Computer science students in non-traditional colleges, <a data-v-0eba368d="" href="https://mentorsintech.com/mentors" target="_blank">MinT</a></li>
        <li data-v-0eba368d=""> Team lead for student capstone project, <a data-v-0eba368d="" href="https://mentorsintech.com/open-source-1" target="_blank">CodeDay.org</a>, <a data-v-0eba368d="" href="https://www.codeday.org/volunteer" target="_blank">CodeDay.org</a>, Green River College </li>
        <li data-v-0eba368d=""> Mentor, MicroInternship <a data-v-0eba368d="" href="https://mentorsintech.com/open-source-1" target="_blank">CodeDay.org</a></li>
        <li data-v-0eba368d=""> General Volunteering, Website and SEO consulting, <a data-v-0eba368d="" href="https://hugohouse.org/" target="_blank">Hugo House</a></li>
        <li data-v-0eba368d="">Instructor, Discrete Math and Data Structures, Northeastern University, Seattle</li>
        <li data-v-0eba368d="">Volunteer Teacher, Introduction to Computer Science, Truman High School, Federal Way</li>
        <li data-v-0eba368d=""> Instructor, Introduction to Programming Workshop, General Assembly, Seattle, Washington </li>
      </ul>
      <div data-v-0eba368d="" class="section">Writing</div>
      <ul data-v-0eba368d="">
        <li data-v-0eba368d=""> Technical Blog: <a data-v-0eba368d="" href="https://medium.com/@dwgray/" target="_blank">https://medium.com/@dwgray/</a></li>
        <li data-v-0eba368d=""> Marketing Blog: <a data-v-0eba368d="" href="https://music4dance.blog/" target="_blank">https://music4dance.blog/</a></li>
      </ul>
    </div>
  </div>
  <div id="footer-content">
    <hr>
    <footer>
      <p> © 2025 - <a href="https://www.music4dance.net">Music4Dance.net</a> - <a href="https://www.music4dance.net/home/sitemap">Site Map</a> - <a href="https://www.music4dance.net/home/termsofservice">Terms of Service</a> - <a href="https://www.music4dance.net/home/privacypolicy">Privacy Policy</a> - <a href="https://www.music4dance.net/home/credits">Credits</a> - <a href="https://github.com/music4dance/music4dance" target="_blank">Code</a> - <a>Help</a></p>
    </footer>
  </div>
</div>"
`;
